### ISSUE TITLE
Remove old numba methods

### ISSUE THREAD

**Issue by RemiLehe** [fbpic/fbpic#86]
These methods are superseded by the new threading methods of PR #80 .

Note: in order to be able to still use the old versions of numba (i.e. <0.34), we should do something like: in threading_utilities.py
```python
if numba.version > 0.34: #The syntax maybe a bit different    
    threading_installed = True    
    prange = numba.prange 
else:
    threading_installed = False
    prange = range
```

Update: We still need to remove:
- [ ] The old gathering functions
- [ ] The old deposition functions (assuming they are not needed for the laser antenna)

**Comment by RemiLehe** [comment]
This depends on PR #80 being merged.

**Comment by RemiLehe** [comment]
We still need to remove:
- [ ] The old gathering functions
- [ ] The old deposition functions (assuming they are not needed for the laser antenna)
